Makefile

```diff
@@ -49,7 +49,8 @@ BASE_VERSION = 2.2.0
 
 # 3rd party image version
 # These versions are also set in the runners in ./integration/runners/
-COUCHDB_VER ?= 2.3
+COUCHDB_VER ?= 3.1
+COUCHDB_LEGACY_VER ?= 2.3
```

Comment:
As per the CouchDB doc (which we may not trust 😄 based on what we found out), [the upgrade from 2.3 to 3.1](https://docs.couchdb.org/en/stable/install/upgrading.html#standalone-single-node-upgrades) looks very similar to how the peers are being upgraded.  Another option is to simply start a new CouchDB instance and the peer will build the DB automatically.

Is there a reason for supporting both 2.3 & 3.1? something related to release version or handling risk or avoid statedb rebuild?

For CI and unit-test/integration test, should we need to repeat the run on both CouchDB 2.3 and CouchDB 3.1 to ensure we aren't breaking any backward compatibility in the ledger code?

Reply:
We could drop support for 2.3 altogether in the 2.{2,next} stream, and then a lot of these changes go away. TBH, I would prefer that in 2.{2,next} we say the minimum required level of Couch is 3.1. But Couch 2 has been around for nearly 10 years (so getting users to migrate may not be easy, as 3.x is not as tested), and our code is not backwards compatible if we remove support for 2.3, so that would require a major version change to rip out the creation of the `_users` table (which is required for removing support for 2.3).

Reply:
I'm not a ledger expert so I'll defer to @cendhu, @manish-sethi and @denyeart. However, my recommendation would be
- to only upgrade couch versions with a good reason (e.g. a new feature of the new version of couch is necessary; or the old couch version will no longer receive security updates, or it has reached EoL)
- to only upgrade couch version across (major) releases
- to never support more than one couch version per release

With this in mind, I would recommend (once again, deferring to the expert) not to go into the business of supporting multiple versions at the same time.

Reply:
@ale-linux Agree. There is some bug (http://danluu.com/fsyncgate/) (https://github.com/apache/couchdb/commit/3505281559513e2922484ebf0996a8846dcc0a34) that is fixed only in CouchDB 3.x and we do need them. At the same time, we are not sure about the stability of CouchDB 3.x (new release). We haven't decided yet on supporting multiple CouchDB versions (mostly, we shouldn't).